Lib9c.Policy/Policy/BlockPolicySource.cs

```diff
@@ -278,7 +278,7 @@ public partial class BlockPolicySource
             // Avoid NRE when genesis block appended
             // Here, index is the index of a prospective block that transaction
             // will be included.
-            long index = blockChain.Count > 0 ? blockChain.Tip.Index : 0;
+            long index = blockChain.Count > 0 ? blockChain.Tip.Index + 1 : 0;
```

Comment:
I think it has a possibility to make some blocks having obsoleted actions, invalid. I described the situation I imagined. The Nth block (having obsoleted actions at N index) will be invalid when appending Nth block.

<img width="1402" alt="image" src="https://user-images.githubusercontent.com/26626194/220584778-42338d9f-48cd-4281-bb3d-011bf8ff9553.png">

So I think it should be branched by block index to avoid making some blocks invalid.
```suggestion
            long index = blockChain.Count > 0 ? (blockChain.Tip.Index > SOME_HARDFORK_INDEX ? blockChain.Tip.Index + 1 : blockChain.Tip.Index) : 0;
```

If my assumption was wrong, please let me know it 🙏🏻 .

Reply:
Hmm... I guess I had it wrong. Seems like there is no easy way around this issue then. 😥

Reply:
Okay, the more I think about it, I think we need to do some damage control at this point. 😣